### PR TITLE
feat : Implemented ifftshift and test

### LINKED ISSUE #23801: ifftshift

### PR DESCRIPTION
**Pull request by VaishnaviMudaliar**
# PR Description

## Related Issue

Close #23801

## Checklist

- [X] Did you add a function?
- [X] Did you add the tests?
- [X] Did you run your tests and are your tests passing?
- [X] Did pre-commit not fail on any check?
- [X] Did you follow the steps we provided?

### CODE REVIEW

**Comment by VaishnaviMudaliar** [comment]
Hey @jshepherd01 , please do review my PR and let me know if any changes required, thanks 😊

**Comment by jshepherd01** [comment]
Hi @VaishnaviMudaliar, sorry about this but I think I shouldn't have been assigned to this issue as my contract at Ivy finished at the end of last week. As such, I'm unassigning myself now and hopefully someone else will be along shortly to review this properly.

**Comment by VaishnaviMudaliar** [comment]
> Hi @VaishnaviMudaliar, sorry about this but I think I shouldn't have been assigned to this issue as my contract at Ivy finished at the end of last week. As such, I'm unassigning myself now and hopefully someone else will be along shortly to review this properly.

Alright, thanks anyway ☺️

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan please do review my PR and let me know about the further changes needed, thanks!😊

**Comment by bipinKrishnan** [comment]
Hey @VaishnaviMudaliar, could you please fix the conflicts and push again?

**Comment by VaishnaviMudaliar** [comment]
I'll do it right away

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , i have resolved the conflicts, please have a look and let me know if further changes are required, thanks! 😊

**Comment by bipinKrishnan** [comment]
Hey @VaishnaviMudaliar, seems like you've rebased incorrectly. You could see that the test implementation you added is missing some code. Can you please correct it? Let me know if you face any issues. Thanks!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 1]
```diff
@@ -216,6 +216,19 @@ def test_jax_numpy_ifft2(
...
+    fn_tree="jax.numpy.fft.ifftshift",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"), shape=(4,), array_api_dtypes=True
+    ),
+)
+def test_jax_numpy_ifftshift(
+    dtype_and_x, backend_fw, frontend, test_flags, fn_tree, on_device
+):
+    input_dtype, arr = dtype_and_x
+
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:231]: The test function is missing some code here

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 2]
```diff
@@ -243,17 +256,26 @@ def test_jax_numpy_rfft(
...
     input_dtype, x, axis = dtype_input_axis
+
     helpers.test_frontend_function(
         input_dtypes=input_dtype,
         frontend=frontend,
         backend_to_test=backend_fw,
         test_flags=test_flags,
         fn_tree=fn_tree,
         on_device=on_device,
+
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:267]: These changes are not related to the function you added, so you could undo these changes.

**Comment by VaishnaviMudaliar** [comment]
@bipinKrishnan I have added the missing code and made the changes, please do check and let me know 🙂

**Comment by bipinKrishnan** [comment]
Hey, still there are same changes not related to your changes, you could check this in the "Files changed" tab at the top, please undo these. Also, you should install pre-commit in your system so that the linting is done automatically before pushing the code, currently the lint tests are failing due to not doing this. Thanks!

**Comment by VaishnaviMudaliar** [comment]
Hi! @bipinKrishnan  , I have run the pre-commit and also removed the unwanted code, please let me know if anymore changes are need . Thanks!

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , please do review the PR and let me know if any more changes are needed, thanks!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 3]
```diff
@@ -269,6 +293,7 @@ def test_jax_numpy_rfft(
     dtype_input_axis, n, norm, frontend, backend_fw, test_flags, fn_tree, on_device
 ):
     input_dtype, x, axis = dtype_input_axis
+
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:296]: Could you please remove this extra line added here as it is not related to the function are working on?

Comment (VaishnaviMudaliar) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:296]: I'll do it right away

Comment (VaishnaviMudaliar) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py]: I have removed the extra code, please check,
Thanks for your patience 😊

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , while making the above mentioned changes , I had accidentally deleted the code for an already implemented function , I apologize for doing so , however , i have added the code back . I would request you to please check the PR . Thanks for your patience :)

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , please do review the PR for the made changes, thanks!😊

File: ivy/functional/frontends/jax/numpy/fft.py [hunk 4]
```diff
@@ -70,6 +70,33 @@ def ifft2(a, s=None, axes=(-2, -1), norm=None):
...
+    # Check if an array
+    if not ivy.is_array(x):
+        raise ValueError("Input 'x' must be an  array")
+
+    # Get the shape of x
+    shape = ivy.shape(x)
+
+    # If axes is None, shift all axes
+    if axes is None:
+        axes = list(range(ivy.ndims(x)))
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/jax/numpy/fft.py:84]: The tests are failing because ivy doesn't have `ndims` attribute.

Comment (VaishnaviMudaliar) [reply ivy/functional/frontends/jax/numpy/fft.py]: I guess I missed that while reading the docs , however , I have changed it to len(shape)

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , I have changed the code as suggested, I would request you to please review the PR, thanks!

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , please do review the PR for the changes made! Thank you!😊

**Comment by VaishnaviMudaliar** [comment]
Hey @bipinKrishnan , please do review the PR for the changes made! Thank you!😊

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
